libkbfs/folder_branch_ops.go

```diff
@@ -6852,12 +6861,178 @@ func (fbo *folderBranchOps) KickoffAllOutstandingRekeys() error {
 func (fbo *folderBranchOps) NewNotificationChannel(
 	ctx context.Context, handle *TlfHandle, convID chat1.ConversationID,
 	channelName string) {
-	// TODO(KBFS-2996): invalidate the local edit notification
-	// tracker, and register for notifications of new messages on the
-	// channel.
+	fbo.editActivity.Add(1)
+	fbo.editChannels <- editChannelActivity{convID, channelName, ""}
 }
 
 // PushConnectionStatusChange pushes human readable connection status changes.
 func (fbo *folderBranchOps) PushConnectionStatusChange(service string, newStatus error) {
-	fbo.config.KBFSOps().PushConnectionStatusChange(service, newStatus)
+	switch service {
+	case KeybaseServiceName, GregorServiceName:
+	default:
+		return
+	}
+
+	if newStatus == nil {
+		fbo.log.CDebugf(nil, "Asking for an edit re-init after reconnection")
+		fbo.editActivity.Add(1)
+		fbo.editChannels <- editChannelActivity{nil, "", ""}
+	}
+}
+
+func (fbo *folderBranchOps) receiveNewEditChat(
+	convID chat1.ConversationID, message string) {
+	fbo.editActivity.Add(1)
+	fbo.editChannels <- editChannelActivity{convID, "", message}
```

Comment:
What happens if we reach 100 `editChannels`? This will block...

Reply:
> What happens if we reach 100 editChannels? This will block...

It will block when a single FBO has 100 separate unprocessed edit notifications in the queue, but only until it finishes processing one.  The blocking has to happen somewhere, but I chose 100 pretty arbitrarily.  Let me know if you have a number or blocking plan you think is more reasonable.

Reply:
Naw that sounds reasonable.